store/src/main/java/org/apache/rocketmq/store/index/IndexFile.java

```diff
@@ -145,12 +145,9 @@ public boolean putKey(final String key, final long phyOffset, final long storeTi
         return false;
     }
 
-    public int indexKeyHashMethod(final String key) {
+    private int indexKeyHashMethod(final String key) {
         int keyHash = key.hashCode();
-        int keyHashPositive = Math.abs(keyHash);
```

Comment:
Will hashcode equal the minimum value of int?

Reply:
Oh, you remind me, this case can lead to serious problem.